org.eclipse.m2e.tests/src/org/eclipse/m2e/tests/ProjectConfigurationManagerTest.java

```diff
@@ -570,6 +570,37 @@ public void testHiddenFolderForArchetypeModule() throws Exception {
     }
   }
 
+  public void test486737_projectUpdate() throws Exception {
+    IProject project = importProject("projects/486737_lifecycleParticipant/pom.xml");
+    waitForJobsToComplete();
+    assertNoErrors(project);
```

Comment:
I think it'd be better to read a pom not declaring the extension 1st, check source folders are src/main/java|resources, then switch the pom to activate the extension, update and check the new src folder is declared